plugins/modules/azure_rm_tags.py

```diff
@@ -0,0 +1,233 @@
+    operation: Merge
+    tags:
+      key2: value2
+
+- name: Delete the tags by scope
+  azure_rm_tags:
+    scope: "/subscriptions/xxxxxxxxxxxxxxxxxxxxxxxxxx/resourceGroups/v-xisuRG02"
+    state: absent
+'''
+RETURN = '''
+tag_info:
+    description:
+        - The tag info.
+    returned: when I(scope=None)
+    type: complex
+    contains:
+        id:
+            description:
+                - The ID of the tags wrapper resource.
+            returned: always
+            type: str
+            sample: "/subscriptions/xxx-xxx/resourceGroups/testRG/providers/Microsoft.Resources/tags/default"
+        name:
+            description:
+                - The name of the tags wrapper resource.
+            returned: always
+            type: str
+            sample: default
+        type:
+            description:
+                - The type of the tags wrapper resource.
+            returned: always
+            type: str
+            sample: Microsoft.Resources/tags
+        properties:
+            description:
+                - The set of tags.
+            returned: always
+            type: dict
+            sample: { 'tags': {'key1': 'value1', 'key2': 'value2'}}
+'''
+
+
+try:
+    from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+    import copy
+    from azure.core.polling import LROPoller
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMTags(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            scope=dict(type='str', required=True),
+            operation=dict(type='str', choices=['Replace', 'Merge', 'Delete'], default='Merge'),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+        )
+
+        self.scope = None
+        self.operation = None
+        self.tags = None
+        self.state = None
+
+        self.results = dict(
+            changed=False,
+            tag_info=dict()
+        )
+
+        super(AzureRMTags, self).__init__(self.module_arg_spec,
+                                          supports_tags=True,
+                                          supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        changed = False
+        if self.state == 'present':
+            response = self.get_at_scope()
+            if response and response['properties'].get('tags'):
+                if self.tags:
+                    update_tags = self.tags_update(response['properties']['tags'], self.tags)
+
+                    if (self.operation == 'Merge' or self.operation == 'Replace') and update_tags:
+                        changed = True
+                        response = self.begin_update_at_scope(self.tags, self.operation)
+                    elif not update_tags and self.operation == 'Delete':
```

Comment:
Instead of testing on `not update_tags`, shoulnd't we just test if the remote has non-empty tags, which is already checked at line 146?